parse/MovableEnvelope.h

```diff
@@ -203,6 +208,20 @@ namespace parse { namespace detail {
         { return MovableEnvelope<T>(obj); }
     };
 
+    /** \p get_pointer is a functor that returns the original object of a MovableEnvelope<T> */
+    struct get_pointer {
+        template <typename T>
+        using result_type = const T*;
```

Comment:
why is `result_type` needed?

Reply:
for boost result_type protocol [Boost.Result Of](https://www.boost.org/doc/libs/1_74_0/libs/utility/utility.htm#result_of). but actually i do not need get_pointer anymore, so i will remove it.